src/main/scala/uk/gov/hmrc/play/auth/controllers/AuthConfig.scala

```diff
@@ -81,11 +58,14 @@ trait AuthParamsControllerConfig {
       account = config.getAs[String]("account"),
       agentRole = config.getAs[String]("agentRole"),
       delegatedAuthRule = config.getAs[String]("delegatedAuthRule"),
-      levelOfAssurance = config.getAs[LevelOfAssurance]("levelOfAssurance").getOrElse(globalLOA)
+      confidenceLevel = config.getAs[Int]("confidenceLevel").orElse(GlobalConfidenceLevel).getOrElse(throw NoConfidenceLevelError)
     )
```

Comment:
Stacktraces get generated when an Exception is instantiated, not when it is thrown. It's therefore better not to reuse existing instances as their stacktrace might be wrong.